pkg/adapter/metrics.go

```diff
@@ -69,7 +69,19 @@ type (
 		Builder
 
 		// NewMetrics returns a new instance of the Metrics aspect.
-		NewMetrics(env Env, config AspectConfig) (MetricsAspect, error)
+		NewMetrics(env Env, config AspectConfig, metrics []MetricDefinition) (MetricsAspect, error)
+	}
+
+	// MetricDefinition provides the basic description of a metric schema
+	// for which metrics adapters will be sent Values at runtime.
+	MetricDefinition struct {
+		// Name is the canonical name of the metric.
+		Name string
+		// Kind provides type information about the metric.
+		Kind Kind
+		// Labels are the names of keys for dimensional data that will
+		// generated at runtime and passed along with metric values.
+		Labels []string
```

Comment:
+1, I think having the label's type on hand is valuable. Also, have we spent any time thinking about how labels are created/defined? This is relevant for the statsd adapter I'm working on: statsd metrics are dimensionless so the common pattern is to encode dimensions in the metric's name. Given the current interface, these dimensions would have to be passed into the aspect impl as labels.